packages/plugins/src/hardhat/deployments.ts

```diff
@@ -336,23 +339,34 @@ export const proposeChugSplashBundle = async (
   configUri: string,
   remoteExecution: boolean,
   ipfsUrl: string,
-  isUpgrade: boolean,
-  configPath: string
+  configPath: string,
+  spinner: ora.Ora,
+  confirm: boolean
 ) => {
+  // Determine if the deployment is an upgrade
+  const isUpgrade = await checkIsUpgrade(hre.ethers.provider, parsedConfig)
   if (isUpgrade) {
+    spinner.succeed('Upgrade detected')
+    // Check if upgrade is valid
     await checkValidUpgrade(
       hre.ethers.provider,
       parsedConfig,
       configPath,
       hre.network.name
     )
-  } else {
-    await checkValidDeployment(
-      hre.ethers.provider,
-      parsedConfig,
-      configPath,
-      hre.network.name
-    )
+
+    if (!confirm) {
+      // Confirm upgrade with user
+      const userConfirmed = await yesno({
+        question: `Prior deployments detected for project ${parsedConfig.options.projectName}, would you like to perform an upgrade? (y/n)`,
+      })
+      if (!userConfirmed) {
+        throw new Error(
+          `User denied upgrade. The project name ${parsedConfig.options.projectName} was already used in a previous deployment. To perform a
+  fresh deployment of a new project, you must change the project name to something other than ${parsedConfig.options.projectName}.`
+        )
```

Comment:
The original error message inside `checkValidDeployment` [here](https://github.com/chugsplash/chugsplash/blob/develop/packages/core/src/utils.ts#L106-L114) also suggested that the user could change the reference name to something other than ${referenceName}. Would mind adding that in?